src/components/molecules/AssetStatus.tsx

```diff
@@ -0,0 +1,51 @@
+import { DID } from '@oceanprotocol/lib'
+import axios, { CancelTokenSource } from 'axios'
+import React, { ReactElement, useEffect, useState } from 'react'
+import { useOcean } from '../../providers/Ocean'
+import { checkFile } from '../../utils/provider'
+import Status from '../atoms/Status'
+import Tooltip from '../atoms/Tooltip'
+import styles from './AssetStatus.module.css'
+
+export default function AssetStatus({
+  did
+}: {
+  did: DID | boolean
+}): ReactElement {
+  const { ocean } = useOcean()
+  const [isFileValid, setFileValid] = useState<boolean>()
+
+  useEffect(() => {
+    async function validateAsset(did: DID, source: CancelTokenSource) {
+      const ddo = await ocean.metadataCache.retrieveDDO(did)
+      if (!ddo) return
+
+      const fileValid = await checkFile(
+        DID.parse(ddo.id),
```

Comment:
so the component accepts a `did` but then fetches the DDO again just to get the `did`?